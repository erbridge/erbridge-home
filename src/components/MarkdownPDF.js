import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { getAsset } from '../assets';

import './MarkdownPDF.scss';

class MarkdownPDF extends Component {
  static propTypes = {
    paperSize: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  };

  render() {
    const { paperSize, src, title } = this.props;

    return (
      <p className={classnames('MarkdownPDF', `MarkdownPDF--${paperSize}`)}>
        <iframe
          className="MarkdownPDF__content"
          title={title}
          src={getAsset(src)}
          frameBorder="0"
        />
      </p>
    );
  }
}

export default MarkdownPDF;
