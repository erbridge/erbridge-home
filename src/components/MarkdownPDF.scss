@import '../styles/_layout.scss';

.MarkdownPDF {
  width: 100%;
  max-height: 60vh;
}

.MarkdownPDF--A4 {
  @include aspect-ratio(MarkdownPDF__content, 210, 297);
}
