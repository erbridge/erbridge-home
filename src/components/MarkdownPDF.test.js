import { shallow } from 'enzyme';
import React from 'react';

import MarkdownPDF from './MarkdownPDF';

it('renders correctly', () => {
  expect(
    shallow(
      <MarkdownPDF title="Example" src="/assets/example.pdf" paperSize="A4" />,
    ),
  ).toMatchSnapshot();
});
