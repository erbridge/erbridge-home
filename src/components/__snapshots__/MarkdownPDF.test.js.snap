// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="MarkdownPDF MarkdownPDF--A4"
>
  <iframe
    className="MarkdownPDF__content"
    frameBorder="0"
    src="/assets/example.pdf"
    title="Example"
  />
</p>
`;
